### PR TITLE
add extract domain/service/role name functions to AthenzUtils

### PR DESCRIPTION
**Pull request by havetisyan**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: libs/java/auth_core/src/main/java/com/yahoo/athenz/auth/util/AthenzUtils.java [hunk 1]
```diff
@@ -90,4 +90,66 @@ public static boolean isRoleCertificate(X509Certificate x509Cert) {
...
+        } else {
+            return roleName.substring(0, idx);
+        }
+    }
+
+    /**
+     * Extract the domain name from the principal name which
+     * is in the format {domain}.{service}
+     * @param principalName full name of the principal
+     * @return domain name, null if it's not the expected full service name
```
Comment (abvaidya) [inline libs/java/auth_core/src/main/java/com/yahoo/athenz/auth/util/AthenzUtils.java]: if it's not the expected full domain name

File: libs/java/auth_core/src/main/java/com/yahoo/athenz/auth/util/AthenzUtils.java [hunk 2]
```diff
@@ -90,4 +90,66 @@ public static boolean isRoleCertificate(X509Certificate x509Cert) {
...
+            return roleName.substring(idx + ROLE_SEP.length());
+        }
+    }
+
+    /**
+     * Extract the domain name from the full Athenz Role Name (arn)
+     * which includes the role name. The format of the role name
+     * is {domain}:role.{role-name}
+     * @param roleName the full arn of the role
+     * @return role name, null if it's not expected full arn format
```
Comment (abvaidya) [inline libs/java/auth_core/src/main/java/com/yahoo/athenz/auth/util/AthenzUtils.java]: @return domain name

Comment (havetisyan) [reply libs/java/auth_core/src/main/java/com/yahoo/athenz/auth/util/AthenzUtils.java]: will fix it now

File: servers/zts/src/main/java/com/yahoo/athenz/zts/cache/DataCache.java [hunk 3]
```diff
@@ -17,9 +17,9 @@
 
 import java.util.*;
```
Comment (pratikgote) [inline servers/zts/src/main/java/com/yahoo/athenz/zts/cache/DataCache.java]: can we expand the * ?

File: servers/zts/src/main/java/com/yahoo/athenz/zts/cache/DataCache.java [hunk 4]
```diff
@@ -17,9 +17,9 @@
...
 
+import com.yahoo.athenz.auth.util.AthenzUtils;
 import com.yahoo.athenz.zms.*;
```
Comment (pratikgote) [inline servers/zts/src/main/java/com/yahoo/athenz/zts/cache/DataCache.java]: can we expand the * ?

**Comment by havetisyan** [comment]
@pratikgote IntelliJ automatically switches to * when it determines that we're doing too many imports. So I've been ok with it as I would expect the default behavior is most likely what's correct for most projects and no need to keep overriding it.

**Pull request merged**